sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/rx/ParallelDocumentQueryTest.java

```diff
@@ -718,6 +718,7 @@ public void readMany() throws Exception {
         assertThat(documentFeedResponse.getResults().size()).isEqualTo(itemIdentities.size());
         assertThat(documentFeedResponse.getResults().stream().map(jsonNode -> jsonNode.get("id").textValue()).collect(Collectors.toList()))
             .containsAll(itemIdentities.stream().map(i -> i.getId()).collect(Collectors.toList()));
+        assertThat(documentFeedResponse.getCosmosDiagnostics()).isNotNull();
```

Comment:
This is good, but I think we should try for more detailed tests to verify the feedresponse diagnostics and query metrics properly. 
One way would be to add couple of new tests to `CosmosDiagnosticsTest`

Reply:
Added one in aacb081f1da, would love to have more pointers on things to test for.

Reply:
That should be e995b5fa83a now that I did an ammend.

Reply:
New test looks good and adds sufficient coverage